fontforgeexe/cvpalettes.c

```diff
@@ -4556,19 +4568,28 @@ void BVPaletteSetVisible(BitmapView *bv,int which,int visible) {
     SavePrefs(true);
 }
 
-void BVPaletteActivate(BitmapView *bv) {
+static void _BVPaletteActivate(BitmapView *bv, int force) {
     BitmapView *old;
 
     BVPaletteCheck(bv);
-    if ( (old = GDrawGetUserData(bvtools))!=bv ) {
+    if ( ((old = GDrawGetUserData(bvtools)) != bv) || force ) {
 	if ( old!=NULL ) {
 	    SaveOffsets(old->gw,bvtools,&bvtoolsoff);
 	    SaveOffsets(old->gw,bvlayers,&bvlayersoff);
 	    SaveOffsets(old->gw,bvshades,&bvshadesoff);
 	}
+
```

Comment:
The double `BVPaletteCheck()` seems even more superfluous here. `GDrawGetUserData()` is NULL-tolerant so just move the initial call to an `else`.

Reply:
At a cursory glance that seems so